### ISSUE TITLE
getStreamInfo() is deleted that breaks compatibility

### ISSUE THREAD

**Issue by crazyzhou** [pravega/pravega#6850]
**Describe the bug**
In #6748, it is decides that to rename the `getStreamInfo()` to `fetchStreamInfo()` and change the signature of the method. This change breaks the backward compatibility guarantee, thus causes the Spark connector compile failure.
There are some `getStreamInfo` usages like here:
https://github.com/pravega/spark-connectors/blob/bd1bb8acec3e12e765e8ef977a6278204919a37e/src/main/scala/io/pravega/connectors/spark/PravegaMicroBatchStream.scala#L58

**To Reproduce**
Update the Pravega to the latest and compile the Spark connector

**Additional information**
The connector can update, but as there may be other `getStreamInfo` usages in other projects and we promised the backward compatibility, I would suggest we add the method back, maybe something like this.
```
StreamInfo getStreamInfo(String scopeName, String streamName) {
    return fetchStreamInfo(scopeName, streamName).get();
}
```

**Comment by ShwethaSNayak** [comment]
Java documentation update for more clarity

**Comment by RaulGracia** [comment]
Closed via #6873.
